internal/validation/mock_trace_validation.go

```diff
@@ -18,55 +18,86 @@ import (
 	"context"
 	"fmt"
 	"reflect"
+	"regexp"
 	"sync"
 	"time"
 
 	"github.com/golang/protobuf/ptypes"
 
 	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
 	"google.golang.org/genproto/googleapis/rpc/errdetails"
-	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+const (
+	maxAnnotationAttributes = 4
+	maxAnnotationBytes      = 256
+	maxAttributes           = 32
+	maxAttributeKeyBytes    = 128
+	maxAttributeValueBytes  = 256
+	maxDisplayNameBytes     = 128
+	maxLinks                = 128
+	maxTimeEvents           = 32
+)
+
 var (
-	ErrInvalidTimestamp   = status.Error(codes.InvalidArgument, "start time must be before end time")
-	ErrMalformedTimestamp = status.Error(codes.InvalidArgument, "unable to parse timestamp")
-	ErrDuplicateSpanName  = status.New(codes.AlreadyExists, "duplicate span name")
-	requiredFields        = []string{"Name", "SpanId", "DisplayName", "StartTime", "EndTime"}
+	requiredFields   = []string{"Name", "SpanId", "DisplayName", "StartTime", "EndTime"}
+	spanNameRegex    = regexp.MustCompile("^projects/(.*)/traces/[a-fA-F0-9]{32}/spans/[a-fA-F0-9]{16}")
+	projectNameRegex = regexp.MustCompile("^projects/(.*)")
```

Comment:
As above. Why is there a capturing group? You don't need parenthesis around `.*` if you aren't pulling the values out.